### ISSUE TITLE
Support Azure Role Assignments on Management Groups

### ISSUE THREAD

**Issue by stefan-rapp** [terraform-providers/terraform-provider-azurerm#4040]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.6
provider.azurerm v1.32.1

### Affected Resource(s)

azurerm_role_assignment
azurerm_role_definition

### Terraform Configuration Files

```
resource "azurerm_role_assignment" "role_assignment" {
  name                  = "d9e21632-402a-4215-b483-9aa9ac658f9a"
  scope                 = "/providers/Microsoft.Management/managementGroups/myMG"
  role_definition_id    = "8e3af657-a8ff-443c-a75c-2fe8c4bcb635" #Owner Role (built-in)
  principal_id          = "d7341423-f174-480c-923b-16b330f3c672"
}
```

### Expected Behavior

Role assignment to the Azure Management Group, according to the code above.

### Actual Behavior

`Error: authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequestFormat" Message="The request was incorrectly formatted."`

### References

https://github.com/terraform-providers/terraform-provider-azurerm/issues/1761

**Comment by nexxai** [comment]
Submitted a PR to add support for Management Groups

**Comment by stefan-rapp** [comment]
Thx guys!

**Comment by tombuildsstuff** [comment]
Fixed via #4063 - thanks again @nexxai :)
